ChartsDemo/Swift/Demos/CombinedChartViewController.swift

```diff
@@ -92,28 +92,27 @@ class CombinedChartViewController: DemoBaseViewController {
     }
     
     override func optionTapped(_ option: Option) {
+        guard let data = chartView.data else
+        {
+            super.handleOption(option, forChartView: chartView)
+            return
+        }
+
         switch option {
         case .toggleLineValues:
-            for set in chartView.data!.dataSets {
-                if let set = set as? LineChartDataSet {
-                    set.drawValuesEnabled = !set .drawValuesEnabled
-                    
-                }
-            }
+            data.dataSets.filter { $0 is LineChartDataSet }
+                         .forEach { $0.drawValuesEnabled = !$0.drawValuesEnabled }
```

Comment:
This is fine here, but if you plan on contributing to the Charts framework avoid this. Here we have to loop through `dataSets` twice to enable toggle `drawValuesEnabled`, whereas something like :
```
for case let set as LineChartDataSet in data.dataSets {
    set.drawValuesEnabled = !set.drawValuesEnabled
}
```
only has to loop once.
I am curious what the performance characteristics of:
```
data.dataSets.lazy
    .flter { ... }
    .forEach { ... }
```
are like.

Reply:
I don't know an answer for your last question. 
Definitely, `sequence.filter{ $0.condition }.forEach{ $0.doSomething }` is slower than `sequence.forEach{ if $0.condition { $0.doSomething } }`. One is faster, another is readable. I guess, here, where we loop over data sets, minor performance loss could be sacrificed for readability.